File: utils/threads-common/src/main/java/com/frolo/threads/HandlerExecutor.java
package com.frolo.threads;

import android.os.Handler;
import android.os.Looper;

import androidx.annotation.NonNull;

import java.util.concurrent.Executor;


public class HandlerExecutor implements Executor {
    @NonNull
    public static HandlerExecutor main() {
        return new HandlerExecutor(Looper.getMainLooper());
    }

    @NonNull
    private final Handler mHandler;

    public HandlerExecutor(@NonNull Handler handler) {
        mHandler = handler;
    }

    public HandlerExecutor(@NonNull Looper looper) {
        this(new Handler(looper));
    }

    @Override
    public void execute(Runnable command) {
        mHandler.post(command);
    }
}

File: app/src/main/java/com/frolo/muse/HandlerExecutor.kt
package com.frolo.muse

import android.os.Handler
import android.os.Looper
import java.util.concurrent.Executor


class HandlerExecutor(private val handler: Handler): Executor {

    constructor(looper: Looper):  this(Handler(looper))

    override fun execute(command: Runnable?) {
        if (command != null) {
            handler.post(command)
        }
    }

}